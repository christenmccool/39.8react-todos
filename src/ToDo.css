.ToDo {
  margin-bottom: 30px;
}

.ToDo-main {
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  gap: 10px;
}

.ToDo-task {
  margin: 2px 10px;
}

.ToDo-task-complete {
  text-decoration: line-through;
  color: darkgrey;
}

.ToDo-task-high {
  font-weight: bold;
}

.ToDo-task-low {
  font-weight: lighter;
}

.ToDo-delete-btn, .ToDo-edit-btn, .ToDo-complete-btn {
  border-radius: 5px;
  padding: 1px 8px;
}

.ToDo-complete-btn {
  border: 1px solid darkgrey;
  color: darkgrey;
  background-color: white;
  color: black;
  border-weight: bold;
}

.ToDo-delete-btn {
  border: none;
  background-color: black;
  color: white;
  border-weight: bold;
}

.ToDo-edit-btn {
  border: none;
  background-color: darkgrey;
  color: black;
  border-weight: bold;
}

.ToDo-fadeout{
  animation: fadeout 2s linear forwards;
}
  
@keyframes fadeout {
  from { opacity: 1; }
  to { opacity: 0; }
}



