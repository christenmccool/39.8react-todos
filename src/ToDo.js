import React from 'react';
import './ToDo.css';
import EditToDoForm from './EditToDoForm';

const ToDo = ( {id, task, priority, complete, toEdit, fadeout, deleteToDo, toggleProp, editToDo} ) => {
  let taskClassName = complete ? "ToDo-task ToDo-task-complete" : "ToDo-task";
  taskClassName = priority==="high" ? `${taskClassName} ToDo-task-high` : taskClassName;
  taskClassName = priority==="low" ? `${taskClassName} ToDo-task-low` : taskClassName;
  taskClassName = priority==="low" ? `${taskClassName} ToDo-task-low` : taskClassName;
  
  let ToDoclassName = fadeout ? "ToDo-fadeout" : "ToDo";

  const formStyle = toEdit ? {display: 'block'} : {display: 'none'};

  const handleDelete = () => {
    toggleProp("fadeout");
    setTimeout(() => deleteToDo(), 2500);
    // deleteToDo();
  }

  return (
    <div className={ToDoclassName}>
      <div className="ToDo-main">
        <div className={taskClassName}>{task}</div>
        
        <button className="ToDo-complete-btn" onClick={()=>toggleProp('complete')}>
          Complete
        </button>

        <button className="ToDo-delete-btn" onClick={handleDelete}>
          X
        </button>

        <button className="ToDo-edit-btn" onClick={()=>toggleProp('toEdit')}>
          Edit
        </button>
      </div>

      <div className="ToDo-editform" style = {formStyle}>
        <EditToDoForm 
          id={id}
          task={task}
          priority={priority}
          toggleProp={toggleProp}
          editToDo={editToDo}
        />
      </div>
    </div>
  )
}

export default ToDo;