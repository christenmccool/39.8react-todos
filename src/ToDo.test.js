import { render, screen } from '@testing-library/react';
import ToDo from './ToDo';

test('if renders without crashing', () => {
  render(<ToDo />);
});

test("if matches snapshot", () => {
  const {asFragment} = render(<ToDo />);
  expect(asFragment()).toMatchSnapshot();
})

