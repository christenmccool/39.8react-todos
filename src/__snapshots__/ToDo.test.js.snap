// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToDo"
  >
    <div
      class="ToDo-main"
    >
      <div
        class="ToDo-task"
      />
      <button
        class="ToDo-complete-btn"
      >
        Complete
      </button>
      <button
        class="ToDo-delete-btn"
      >
        X
      </button>
      <button
        class="ToDo-edit-btn"
      >
        Edit
      </button>
    </div>
    <div
      class="ToDo-editform"
      style="display: none;"
    >
      <form
        class="EditToDoForm"
      >
        <label
          class="EditToDoForm-label"
          for="task"
        >
          Edit task
        </label>
        <input
          id="task"
          name="task"
          type="text"
          value=""
        />
        <fieldset
          class="EditToDoForm-priority"
        >
          <legend>
            Priority
          </legend>
          <input
            id="choice3"
            name="priority"
            type="radio"
            value="low"
          />
          <label
            class="EditToDoForm-label"
            for="choice3"
          >
            Low
          </label>
          <input
            id="choice2"
            name="priority"
            type="radio"
            value="medium"
          />
          <label
            class="EditToDoForm-label"
            for="choice2"
          >
            Medium
          </label>
          <input
            id="choice1"
            name="priority"
            type="radio"
            value="high"
          />
          <label
            class="EditToDoForm-label"
            for="choice1"
          >
            High
          </label>
        </fieldset>
        <button
          class="EditToDoForm-btn"
        >
          Save
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
